packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -83,142 +88,69 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-                )
-            })?;
 
-            let pkg = pkg.run(&mut try_catch).ok_or_else(|| {
-                let exception = try_catch.exception().unwrap();
-                let exception_string = exception
-                    .to_string(&mut try_catch)
-                    .unwrap()
-                    .to_rust_string_lossy(&mut try_catch);
+        let namespace: Object<'_> = {
+            let pkg = match ModuleMap::import_module(scope, &path) {
+                Ok(s) => s,
+                Err(Error::MissingJavascriptImport(_)) => {
+                    tracing::debug!(
+                        "Couldn't read package file. It might intentionally not exist."
+                    );
+                    // Packages don't have to use JS.
+                    let undefined = v8::undefined(scope).into();
+                    let fns = v8::Array::new_with_elements(scope, &[undefined; 3]);
 
-                Error::PackageImport(
-                    path.clone(),
-                    format!("Couldn't execute package setup: {exception_string}"),
-                )
-            })?;
+                    return Ok(JsPackage { fns });
+                }
+                Err(err) => return Err(err),
+            };
 
-            pkg.try_into().map_err(|err| {
-                Error::PackageImport(
-                    path.clone(),
-                    format!("Could not convert package setup return value into Function: {err}"),
-                )
-            })?
+            pkg.get_module_namespace()
+                .to_object(scope)
+                .expect("Module is not instantiated")
         };
 
-        let args = &[embedded.hash_util, embedded.hash_stdlib];
-        let fns: Array<'_> = {
-            let global_context = context.global(scope);
-            let fns = call_js_function(scope, pkg, global_context.into(), args).map_err(|err| {
-                Error::PackageImport(path.clone(), format!("Couldn't call package: {err}"))
-            })?;
-
-            fns.try_into().map_err(|err| {
-                Error::PackageImport(
-                    path.clone(),
-                    format!(
-                        "Couldn't convert package return value to array of package functions: \
-                         {err}"
-                    ),
-                )
-            })?
-        };
+        let fn_names = ["start_experiment", "start_sim", "run_task"];
+        let fns = fn_names
+            .into_iter()
+            .map(|fn_name| {
+                let mut try_catch = v8::TryCatch::new(scope);
+
+                let js_fn_name = new_js_string(&mut try_catch, fn_name);
+                // Get the function `Value` from the namespace
+                let func_or_undefined: Value<'_> = namespace
+                    .get(&mut try_catch, js_fn_name.into())
+                    .ok_or_else(|| {
+                        let exception = read_try_catch_exception(&mut try_catch).unwrap();
```

Comment:
This is a little bit repetitive, `read_exection(&mut try_catch_scope)` is good enough I guess. Thoughts?

Reply:
I was aware of `Module::get_exception` and expected more types to have this kind of function.
But it seems it's only `TryCatch` and `Module`.

Renamed in https://github.com/hashintel/hash/pull/493/commits/9b29b4c34a8543ee7a8b065b17fe1875724e8289